### PR TITLE
Darwin stdenv tweaks

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change

Also some miscellaneous tweaks.

### CODE REVIEW

File: pkgs/development/libraries/libuv/default.nix [hunk 1]
```diff
@@ -12,33 +11,45 @@ stdenv.mkDerivation rec {
...
+        "multiple_listen" "delayed_accept"
+        "shutdown_close_tcp" "shutdown_eof" "shutdown_twice" "callback_stack"
+      ];
     tdRegexp = lib.concatStringsSep "\\|" toDisable;
     in lib.optionalString doCheck ''
       sed '/${tdRegexp}/d' -i test/test-list.h
     '';
 
   nativeBuildInputs = [ automake autoconf libtool pkgconfig ];
-  buildInputs = stdenv.lib.optionals stdenv.isDarwin [ ApplicationServices CoreServices ];
```
Comment (LnL7) [inline pkgs/development/libraries/libuv/default.nix]: 👍

File: pkgs/development/libraries/libuv/default.nix [hunk 2]
```diff
@@ -12,33 +11,45 @@ stdenv.mkDerivation rec {
...
     LIBTOOLIZE=libtoolize ./autogen.sh
   '';
 
   enableParallelBuilding = true;
 
-  # These should be turned back on, but see https://github.com/NixOS/nixpkgs/issues/23651
-  # For now the tests are just breaking large swaths of the nixpkgs binary cache for Darwin,
-  # and I'd rather have everything else work at all than have stronger assurance here.
-  doCheck = !stdenv.isDarwin;
+  doCheck = true;
```
Comment (LnL7) [inline pkgs/development/libraries/libuv/default.nix]: I'd also prefer to run them, but dealing with the intermittent failures on hydra is really painful particularly for staging. Let's keep an eye on it, let's hope we can get rid of them like this.

File: pkgs/development/libraries/libuv/make-apple-frameworks-optional.patch [hunk 3]
```diff
@@ -0,0 +1,176 @@
+From 6d03644817fb263489dc9fdf550bf1fac274fd8f Mon Sep 17 00:00:00 2001
```
Comment (LnL7) [inline pkgs/development/libraries/libuv/make-apple-frameworks-optional.patch]: I'd fatchpatch this.

**Review by LnL7 — APPROVED** [review]

File: pkgs/development/libraries/libuv/default.nix (hunk 2 above)
Comment (matthewbauer) [reply pkgs/development/libraries/libuv/default.nix]: Yes - it's very weird that it fails so frequently. FWIW it's not just libuv that has issues on macOS with network failures. Could it be a kernel thing? Or maybe it's just no one runs tests on macOS.

**Comment by periklis** [comment]
@matthewbauer any number of this reduction to get an impression?

**Comment by matthewbauer** [comment]
@periklis Sorry my title was a little misleading - these aren't reducing anything currently just preventing the SDK from getting in some places where it shouldn't be.

File: pkgs/development/libraries/libuv/default.nix (hunk 2 above)
Comment (matthewbauer) [reply pkgs/development/libraries/libuv/default.nix]: I wish we could just drop libuv from cmake altogether - but apparently cmake doesn't support that for some weird reason.

**Pull request merged**